### PR TITLE
#3718 - Implement SSO for rca.ac.uk

### PR DESCRIPTION
**Pull request by patrickcuagan**
Ticket: https://torchbox.monday.com/boards/1472452416/views/14036164/pulses/1582473718

This PR adds SSO to the RCA site. Users can then log in via SSO - if they have a Microsoft AD account.

Additional features:

1. Users who logs in via SSO are automatically added as an "Editor".
2. Users who logs in via SSO will see a logout confirmation page when they logout from the Wagtail admin.

### CODE REVIEW

File: rca/account_management/views.py [hunk 1]
```diff
@@ -164,7 +165,7 @@ def form_valid(self, form):
 class CustomLoginView(LoginView):
     """Custom login view to redirect students to their profile page"""
 
-    template_name = "wagtailadmin/login.html"
+    template_name = "patterns/pages/auth/login.html"
 
     def get_success_url(self):
         """Override the success URL if the user meets the following:
```
Comment (kevinhowbrook) [inline rca/account_management/views.py:171]: We should make sure to test this on the dev site, as in logging in via wagtail as a student still works well as I'm not sure this is unit tested

**Review by kevinhowbrook — APPROVED** [review]
Thanks @patrickcuagan 

Just a few notes but I think this can go up for testing :+1:

File: docs/upgrading.md [hunk 2]
```diff
@@ -104,6 +104,15 @@ As well as testing the critical paths, these areas of functionality should be ch
...
 
+### Users who logged in via SSO automatically gets 'Editor' permissions.
+
+1. When users log in via SSO, they are automatically made editors. This is done via `rca.utils.pipeline.make_sso_users_editors` and added to the `SOCIAL_AUTH_PIPELINE`. RCA will manually elevate permissiosn if necessary. 
+
+### Users who logged in via SSO is redirected to a logout confirmation when they logout.
+
+1. The site overrides the `/admin/logout/` endpoint to redirect users who logged in to `/logout/`. This is a confirmation screen that users will still need to manually log out of their SSO accounts. This is done with `rca.account_management.views.CustomLogoutView` and `rca.account_management.views.SSOLogoutConfirmationView`.
+2. Users who did not log in via SSO should be able to log out without seeing any confirmation screen.
+
```
[the remark this answers is not in the document]
Comment (patrickcuagan) [reply docs/upgrading.md:115]: Will do! I added a more general version for the support runbook.

File: rca/account_management/views.py (hunk 1 above)
Comment (patrickcuagan) [reply rca/account_management/views.py:171]: Yep - I tested this locally and it was working as intended.

**Pull request merged**